src/shogun/base/SGObject.h

```diff
@@ -780,6 +781,28 @@ class CSGObject
 	 */
 	virtual void save_serializable_post() noexcept(false);
 
+	template <ParameterProperties PPVal, typename ...Args, std::enable_if_t<static_cast<bool>(PPVal&ParameterProperties::AUTO)>* = nullptr>
+	void declare_auto_func(AnyParameter& any_pprop, Args&&... args)
+	{
+		static_assert(sg_is_any_base_of<params::AutoInit, Args...>::value,
+				"Expected a params::AutoInit function when passing param with"
+				" ParameterProperty::AUTO!");
+		constexpr size_t idx = get_idx_from_pack<params::AutoInit, Args...>::idx;
+		any_pprop.set_init_function(std::move(std::get<idx>(std::forward_as_tuple(args...))));
+	}
+
+	template <ParameterProperties pprop, typename T, typename ...Args>
+	void declare(T* value, const std::string& name, const std::string& description, Args&& ...args)
+	{
+		auto any_pprop = AnyParameterProperties(description, pprop);
+		auto anyp = AnyParameter(make_any_ref(value), any_pprop);
+		if constexpr (static_cast<bool>(pprop&ParameterProperties::AUTO))
+            declare_auto_func<pprop>(anyp, args...);
```

Comment:
just curious.
Is it possible at this point to still register the old parameters?
Because then we could actually merge this pretty soon

Reply:
no because shogun::Parameter is an incomplete type at this point :/

Reply:
Ah yes of course... well that’s fine....